### PR TITLE
[ur] extend ur_usm_alloc_info_t with pool info

### PR DESCRIPTION
**Pull request by igchor**
Closes: https://github.com/oneapi-src/unified-runtime/issues/326

### CODE REVIEW

File: include/ur.py [hunk 1]
```diff
@@ -843,6 +843,7 @@ class ur_usm_alloc_info_v(IntEnum):
     BASE_PTR = 1                                    ## Memory allocation base pointer info
     SIZE = 2                                        ## Memory allocation size info
     DEVICE = 3                                      ## Memory allocation device info
+    POOL = 4                                        ## Memory allocation pool handle
```
Comment (PatKamin) [inline include/ur.py:846]: Did you mean "pool info" in the comment?

Comment (igchor) [reply include/ur.py:846]: Not really. Well, 'info' would be consistent with the other ones, but this option is actually used to obtain a 'handle' to the pool

**Review by pbalcer — APPROVED** [review]

**Comment by pbalcer** [comment]
Please rebase.

**Comment by igchor** [comment]
> Please rebase.

Done.

File: scripts/core/usm.yml [hunk 2]
```diff
@@ -70,6 +70,8 @@ etors:
       desc: "Memory allocation size info"
     - name: DEVICE
       desc: "Memory allocation device info"
+    - name: POOL_HANDLE
```
Comment (kbenzie) [inline scripts/core/usm.yml:73]: The `DEVICE` is also a handle, `POOL` without `_HANDLE` would be consistent with the other property names.

**Review by kbenzie — APPROVED** [review]
LGTM - one comment

File: include/ur.py (hunk 1 above)
Comment (kbenzie) [reply include/ur.py]: I didn't notice this discussion and [commented](https://github.com/oneapi-src/unified-runtime/pull/348#discussion_r1137327335) again on this. I think the *device info* is a mistake and should be *device handle* in the docs.

File: scripts/core/usm.yml (hunk 2 above)
Comment (igchor) [reply scripts/core/usm.yml]: Good point, I changed it back.

**Pull request merged**